aiida_workgraph/web/frontend/src/components/DataNodeTable.js

```diff
@@ -68,6 +69,51 @@ function DataNode() {
         .catch(error => console.error('Error deleting item: ', error));
     };
 
+    const [toDeleteItem, setToDeleteItem] = useState(null);
+    const [showConfirmDeleteModal, setShowConfirmDeleteModal] = useState(false);
+    const [bodyTextConfirmDeleteModal, setBodyTextConfirmDeleteModal] =  useState(<p></p>);
+    // useEffect to ensure this happens after the toDeletItem has been updated by the delete button click
+    useEffect(() => {
+        if (toDeleteItem != null) {
+          fetch(`http://localhost:8000/api/datanode/delete/${toDeleteItem.pk}?dry_run=True`, {
+              method: 'DELETE',
+          })
+          .then(response => response.json())
+          .then(data => {
+              if (data.deleted_nodes.length > 0) {
+                // the node {item.pk} will be always in the list if the deletion is successfull
+                if (data.deleted_nodes.length > 1) {
+                  let formatted_pks = data.deleted_nodes.map((x) => ` PK<${x.toString()}>`);
```

Comment:
I think only showing the number is enough. Otherwise, there is too much text.
```suggestion
                  let formatted_pks = data.deleted_nodes.map((x) => ` ${x.toString()}`);
```

Reply:
I agree that ideally the usage of the `toString` method is better. Reading the message without marking the PKs numbers is not so clear, since you will have numbers that don't correspond to a nodes (like the number of nodes you want to delete). Ideally I would define a class for item that defines the `toString` method. But I would still define in this method to print `PK<item.pk>`

Reply:
Ah okay it is really only about the list at the end. Yes there I can remove it